### PR TITLE
Refactor GCS plugin to extends common abstractions

### PR DESCRIPTION
**Pull request by willyborankin**
1. Refactor GCS plugin to extends common abstractions
2. Bump up ES to version 7.9.1

### CODE REVIEW

File: repository-gcs/src/main/java/io.aiven.elasticsearch.repositories.gcs/GcsSettingsProvider.java [hunk 1]
```diff
@@ -18,69 +18,46 @@
...
+        storageOptionsBuilder
+                .setTransportOptions(
+                        HttpTransportOptions.newBuilder()
+                                .setConnectTimeout(gcsStorageSettings.connectionTimeout())
+                                .setReadTimeout(gcsStorageSettings.readTimeout())
+                                .build())
+                .setHeaderProvider(() -> Map.of(HttpHeaders.USER_AGENT, HTTP_USER_AGENT))
+                .setCredentials(gcsStorageSettings.gcsCredentials());
+
+        return storageOptionsBuilder.build().getService();
```
Comment (ivanyu) [inline repository-gcs/src/main/java/io.aiven.elasticsearch.repositories.gcs/GcsSettingsProvider.java:59]: This used to be inside `Permissions.doPrivileged`. Is it not needed any more?

Comment (willyborankin) [reply repository-gcs/src/main/java/io.aiven.elasticsearch.repositories.gcs/GcsSettingsProvider.java:59]: it is in the line #41 ... i think it is more readable

**Comment by willyborankin** [comment]
@ivanyu could you plz do not merge it so far ... it seems that we have concurrency issue

**Comment by willyborankin** [comment]
@ivanyu there is no issue ... everything is ok

**Review by ivanyu — APPROVED** [review]
Overall LGTM.
I'd only recommend rephrasing the commit message to follow the imperative mood, e.g. `Refactor GCS plugin to extends common abstractions`.

**Comment by willyborankin** [comment]
@ivanyu done

**Review by ivanyu — APPROVED** [review]

**Pull request merged**